src/pybind/mgr/nfs/export.py

```diff
@@ -350,11 +362,18 @@ def _delete_export(
                 export = self._fetch_export(cluster_id, pseudo_path)
 
             if export:
+                exports_count = 0
+                if export.fsal.name == NFS_GANESHA_SUPPORTED_FSALS[0]:
+                    exports_count = self.get_export_count_with_same_fsal(export.fsal.cmount_path,  # type: ignore
+                                                                         cluster_id, export.fsal.fs_name)  # type: ignore
```

Comment:
It ins't totally clear how the export block is refcounted. AFAICT, there are separate export blocks, and during deletion this check scans all exports to see if its the last export block for the district key.

Reply:
Well I've explained this above with one of discussion with @dparmar18 . Deletion of exports stays the same, only clearing `user_id` we need to check count bcz there maybe multiple exports sharing this same client so can't just delete it. Thus only clear it if only 1 export is left which uses this client

Reply:
Fair enough, Could you link the discussion here please?

Reply:
https://github.com/ceph/ceph/pull/54277#discussion_r1608221982